src/frontend/components/PlivoDialer.js

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import {FloatingActionButton, FontIcon} from 'material-ui';
+
+export default class PlivoDialer extends React.Component {
+  static propTypes = {
+    endpointUsername: React.PropTypes.string,
+    endpointPassword: React.PropTypes.string,
+    number: React.PropTypes.string
+  }
+
+  state = {
+    displayed: true,
+    plivoCallInProgress: false,
+    plivoStatusText: null
+  }
+
+  registerCallbacks() {
+    Plivo.onWebrtcNotSupported = () => {
+      this.setState({visible: false})
+    }
+    Plivo.onReady = () => {
+      console.log('PlivoJS is ready to be used')
+    }
+    Plivo.onMediaPermission = () => {
+      console.log('PlivoJS user media access permissions granted')
+    }
+    Plivo.onLogin = () => {
```

Comment:
Suggestion for how the UI should work.  Let me know if this seems too annoying to do and I can just take a shot at it myself. https://gomockingbird.com/projects/1e1d0tb/I7OCJW.  Basically some suggestions on button color, button placement and removing some of the text messages.  The normal call flow is linked, and then check "Error" and "Call failed" for those two states.  The light blue in the second diagram is BernieColors.lightBlue, but not sure that actually shows up well so just a suggestion.  But some kind of color change on the text I think would be nice.